### PR TITLE
Add device delete by id

### PR DESCRIPTION
**Pull request by skeet70**
This also deprecates `deauthorizeDevice`, since `deleteDevice` with no device id argument accomplishes the same goal.

### CODE REVIEW

**Review by BobWall23 — APPROVED** [review]

File: src/shim/sdk/UserSDK.ts [hunk 1]
```diff
@@ -34,30 +34,44 @@ export function rotateMasterKey(passcode: string) {
...
+// TODO: detect if deviceId is the same as the currently operating device and set deletingCurrentDevice to true
+export const deleteDevice = (deviceId?: number) => {
     checkSDKInitialized();
-    const payload: MT.DeauthorizeDevice = {
-        type: "DEAUTHORIZE_DEVICE",
-        message: null,
+    const deletingCurrentDevice = deviceId === undefined;
+    const payload: MT.DeleteDevice = {
+        type: "DELETE_DEVICE",
+        message: deletingCurrentDevice ? undefined : deviceId,
```
Comment (giarc3) [inline src/shim/sdk/UserSDK.ts:54]: I think this is simpler?
```suggestion
        message: deviceId,
```

File: src/shim/sdk/UserSDK.ts [hunk 2]
```diff
@@ -34,30 +34,44 @@ export function rotateMasterKey(passcode: string) {
...
+    if (deletingCurrentDevice) {
+        clearParentWindowSymmetricKey();
+    }
+    return FrameMediator.sendMessage<MT.DeleteDeviceResponse>(payload)
         .map(({message}) => {
-            clearSDKInitialized();
-            return {transformKeyDeleted: message};
+            if (deletingCurrentDevice) {
+                clearSDKInitialized();
+            }
```
Comment (giarc3) [inline src/shim/sdk/UserSDK.ts:65]: Could this just be in the above after clearing the window key?

**Review by giarc3 — APPROVED** [review]
I couldn't think of a good solution for the TODO

File: src/shim/sdk/UserSDK.ts (hunk 1 above)
Comment (skeet70) [reply src/shim/sdk/UserSDK.ts:54]: It is if we can't come up with a solution to the TODO.

File: src/shim/sdk/UserSDK.ts (hunk 2 above)
Comment (skeet70) [reply src/shim/sdk/UserSDK.ts:65]: No, because we need the SDK initialized still to make the delete request that we're sending to the frame here.

**Pull request merged**